api.js

```diff
@@ -1,24 +1,26 @@
 var https = require('https');
 var http = require('http');
 var zlib = require('zlib');
-module.exports = function (u, p, d, s, o) {
+module.exports = function (u, p, sId, d, s, o) {
     var credentials,
         userName = u,
         password = p,
+        sessionId = sId,
```

Comment:
I don't think `sessionId` is necessary, rather if `sId` is present with _direct_ server then we could create a credentials object to be used in requests. Keep in mind this must be the direct server as my.geotab.com would cause InvalidUserException and re-authenticate. Might be more clear to never set in constructor rather always by setCredentials?

Reply:
Sounds good, I'll make the changes

Reply:
On second thought, there is a check below this that checks if password is passed in and if not it throws an error, but you should be able to instantiate an api object with just server, username and session id which is why I wanted to pass in session id here.  Your thoughts?